precompiles/utils/src/data.rs

```diff
@@ -175,6 +180,13 @@ impl EvmDataWriter {
 		self
 	}
 
+	// Write a solidity-style four-byte function selector for the given signature
+	pub fn write_selector(mut self, signature: &[u8]) -> Self {
```

Comment:
I wrote this in my ERC20 PR. This allows to convert Action enums using the selector macro directly.
https://github.com/PureStake/moonbeam/blob/jeremy-balances-erc20-precompile/precompiles/utils/src/data.rs#L182

You can copy it in your PR or get it for free if the ERC20 PR is merged first.

Reply:
I confirm, the `write_selector` in ERC20 PR is better, copy it.